src/objc/abstract/MCOAbstractMessageRendererCallback.mm

```diff
@@ -11,17 +11,34 @@
 #import "MCOHTMLRendererDelegate.h"
 #import "MCOHTMLRendererIMAPDelegate.h"
 #import "MCOUtils.h"
+#import <Objc/runtime.h>
 
 using namespace mailcore;
 
 MCOAbstractMessageRendererCallback::MCOAbstractMessageRendererCallback(MCOAbstractMessage * message, id <MCOHTMLRendererDelegate> rendererDelegate,
                                    id <MCOHTMLRendererIMAPDelegate> rendererIMAPDelegate)
 {
     mMessage = message;
-    mRendererDelegate = rendererDelegate;
+	if (rendererDelegate) {
```

Comment:
I think the `MCOAbstractMessageRendererCallback` should retain the `rendererDelegate`.
Consider the following code:
```objectivec
id<MCOHTMLRendererDelegate> render = [CustomRenderer new];
MCOIMAPMessageRenderingOperation *renderingOperation = [self.IMAPSession htmlBodyRenderingOperationWithMessage:rawMessage
																										folder:folder
																							  rendererDelegate:render];
[renderingOperation start:^(NSString *HTMLString, NSError *error) {
	BLOCK_SAFE_CALLS(completeBlock, HTMLString, error);
}];
```
and this:
```objectivec
- (MCOIMAPMessageRenderingOperation *) htmlBodyRenderingOperationWithMessage:(MCOIMAPMessage *)message
																	  folder:(NSString *)folder
															rendererDelegate:(id<MCOHTMLRendererDelegate>)rendererDelegate
{
	MCOAbstractMessageRendererCallback * htmlCallBack = new MCOAbstractMessageRendererCallback(NULL, rendererDelegate, NULL);
	IMAPMessageRenderingOperation * coreOp = MCO_NATIVE_INSTANCE->htmlRenderingOperation(MCO_FROM_OBJC(IMAPMessage, message), [folder mco_mcString], htmlCallBack);
	htmlCallBack->release();
	return MCO_TO_OBJC(coreOp);
}
```

and this:
```objectivec
void OperationQueue::startThread()
{
    if (mStarted)
        return;
    
    if (mCallback) {
        mCallback->queueStartRunning();
    }
    
    retain(); // (3)
    mQuitting = false;
    mStarted = true;
    pthread_create(&mThreadID, NULL, (void * (*)(void *)) OperationQueue::runOperationsOnThread, this);
    mailsem_down(mStartSem);
}
```

if the `MCOAbstractMessageRendererCallback` do not retain the `rendererDelegate`, the `mRendererDelegate` will be released before `IMAPMessageRenderingOperation::main`.